chapters/statements.tex

```diff
@@ -41,6 +41,12 @@ \subsection{Execution of an algorithm in a model}\label{execution-of-an-algorith
   If at least one element of an array appears on the left hand side of
   the assignment operator, then the complete array is initialized in
   this algorithm section.
+\item
+  A parameter assigned in an initial algorithm, \cref{initialization-initial-equation-and-initial-algorithm},
+  is initialized with its start-value (i.e.\ the value of the start-attribute)
+  provided the start-value can be evaluated before evaluating the initial algorithm
+  (a literal value can always be evaluated); otherwise a default start-value is used
```

Comment:
This doesn't look as simple as I expected.  Why is it necessary to have the initialization conditioned on whether the start-value can be initialized before the initial algorithm?  To me, that just sounds like badly entangled initialization.

Reply:
I agree that it is a badly entangled initialization (i.e. a model issue), and we should state that more clearly. (I had in mind that an iteration variable without start-value normally leads to diagnostics, but then I realized that people cannot read my mind.)

However, it should be rare and even when it used it is not always the case that the start-values are that critical, and there are two possibilities for handling them - either we forbid it or we add a fall-back as I did above.

The problem with forbidding it is that we currently state that we conceptually always initialize variables at the beginning of each algorithm, and leave it to the tools to skip when not needed. If we want to forbid the entangled mess that might be a bit excessive - and we would then need to more clearly define when.

Reply:
I see it differently.  What you propose is in just as much need of a clearly defined _when_ as the error approach, but the error approach has the advantage that any tool that is able to respect applicable `start` attributes should get the same simulation result.

Assuming that we really want to be able to accept these corner case models at all, it becomes a quality of implementation to do the symbolic processing that proves that a nasty `start` doesn't even come into play, but that's a smaller problem than having simulation results where the values coming out of the initialization is a quality of implementation.

Done this way, when one encounters a model developed by another tool where the `start` cannot be evaluated, this should be interpreted as that the other tool was either able to do the evaluation, or that the other tool was able to prove that the `start` doesn't come into play.  A less careful tool would then guess the latter, give a warning about it, and proceed, but this shouldn't be described in (the normative parts of) the specification.

Reply:
But in the relevant cases the ignored start-value is a guess-value, and there are multiple solutions to the initialization problem (on top of the other mess), and we cannot guarantee that tools produce the same solution even for the same guess-value.

Reply:
I think this part should have been excluded when resolving #2640, as we didn't discuss this part at the phone meeting.  I'll make a PR for removing this part, and open an issue to continue the discussion.